source/Playnite.DesktopApp/ViewModels/InstalledGamesViewModel.cs

```diff
@@ -340,24 +340,35 @@ public void ConfirmDialog()
 
         public void SelectExecutable()
         {
-            var path = dialogs.SelectFile("Executable (.exe)|*.exe");
+            var path = dialogs.SelectFile("Executable (.exe,.bat,lnk)|*.exe;*.bat;*.lnk");
             if (string.IsNullOrEmpty(path))
             {
                 return;
             }
+            var program = Common.Programs.GetProgramData(path);
 
-            var productName = FileVersionInfo.GetVersionInfo(path).ProductName;
-            var import = new ImportableProgram(new Program()
+            // Ignore all target files that are not bat or exe files
+            if (!program.Path.EndsWith(".exe", StringComparison.OrdinalIgnoreCase) &&
+                !program.Path.EndsWith(".bat", StringComparison.OrdinalIgnoreCase))
             {
-                Icon = path,
-                Name = string.IsNullOrWhiteSpace(productName) ? new DirectoryInfo(Path.GetDirectoryName(path)).Name : productName,
-                Path = path,
-                WorkDir = Path.GetDirectoryName(path)
-            }, ProgramType.Win32)
+                return;
+            }
+
+            var import = new ImportableProgram(program, ProgramType.Win32)
             {
                 Selected = true
             };
 
+            // Use shortcut name as game name for .lnk shortcuts
+            if (Path.GetExtension(path).EndsWith(".lnk", StringComparison.OrdinalIgnoreCase))

```

Comment:
Calling GetExtension is not necessary since you are checking with EndsWith anyways.

Reply:
resolved in https://github.com/felixkmh/Playnite/commit/f5f7cfdfadfc60fcadb8493435646768b61e68f1